src/block_disagg/block_disagg_addr.c

```diff
@@ -44,36 +133,72 @@ __wti_block_disagg_addr_pack(
+    uint8_t version, version_min;
     const uint8_t *begin;
 
-    p = 0; /* Avoid compiler warnings. */
-
-    /* FIXME-WT-14644: read extensible byte */
     begin = *buf;
-    WT_RET(__wt_vunpack_uint(buf, 0, &p));
-    WT_RET(__wt_vunpack_uint(buf, 0, &l));
-    WT_RET(__wt_vunpack_uint(buf, 0, &s));
-    WT_RET(__wt_vunpack_uint(buf, 0, &c));
+
+    /* Avoid compiler warnings. */
+    base_lsn_delta = flags = lsn = page_id = size = 0;
+    checksum = 0;
+
+    /* Unpack the address version. */
+    WT_RET(__block_disagg_addr_unpack_version(buf, 0, &version, &version_min));
+    if (version_min > WT_BLOCK_DISAGG_ADDR_VERSION)
+        WT_RET_MSG(session, ENOTSUP,
+          "Unsupported disaggregated address cookie version %" PRIu8 ", min %" PRIu8, version,
+          version_min);
+
+    /* Unpack the address cookie. */
+    WT_RET(__wt_vunpack_uint(buf, 0, &page_id));
+    WT_RET(__wt_vunpack_uint(buf, 0, &flags));
+    WT_RET(__wt_vunpack_uint(buf, 0, &lsn));
+    WT_RET(__wt_vunpack_uint(buf, 0, &base_lsn_delta));
+    WT_RET(__wt_vunpack_uint(buf, 0, &size));
+
+    /* Unpack the checksum as a fixed-length 32-bit integer. */
+    WT_RET(__block_disagg_addr_unpack_uint32(buf, 0, &checksum));
+
+    /* Get the base LSN from the delta. */
+    if (lsn < base_lsn_delta)
+        WT_RET_MSG(session, EINVAL,
+          "Disaggregated address cookie LSN %" PRIu64 " is smaller than base LSN delta %" PRIu64,
+          lsn, base_lsn_delta);
+    base_lsn = lsn - base_lsn_delta;
+
+    /* Check the page ID and size. */
+    if (page_id == WT_BLOCK_INVALID_PAGE_ID)
+        WT_RET_MSG(
+          session, EINVAL, "Disaggregated address cookie page ID %" PRIu64 " is invalid", page_id);
+    if (size == 0)
+        WT_RET_MSG(session, EINVAL, "Disaggregated address cookie size %" PRIu32 " is invalid",
+          (uint32_t)size);
+
+    WT_CLEAR(*cookie);
+    cookie->page_id = page_id;
+    cookie->flags = flags;
+    cookie->lsn = lsn;
+    cookie->base_lsn = base_lsn;
+    cookie->size = (uint32_t)size;
+    cookie->checksum = checksum;
 
     /*
-     * Any disagg ID is valid, so use a size of 0 to define an out-of-band value.
+     * Check the address cookie size, but only (1) if we are reading the current version of the
+     * address cookie, and (2) if there are no unsupported flags. If we are reading a new version,
+     * we can't check the size, as more fields could have been added.
      */
-    if (s == 0) {
-        *page_idp = WT_BLOCK_INVALID_PAGE_ID;
-        *lsnp = 0;
-        *sizep = *checksump = 0;
-    } else {
-        *page_idp = p;
-        *lsnp = l;
-        *sizep = (uint32_t)s;
-        *checksump = (uint32_t)c;
-    }
-    if ((size_t)(*buf - begin) != buf_size)
-        return (EINVAL); /* TODO: need a message */
+    unsupported_flags = flags;
+    FLD_CLR(unsupported_flags, WT_BLOCK_DISAGG_ADDR_ALL_FLAGS);
+    if ((size_t)(*buf - begin) != buf_size && version == WT_BLOCK_DISAGG_ADDR_VERSION &&
```

Comment:
Nit: I suggest to reverse the sequence of conditions here so that `&&` short-circuit logic is more straightforward: First check for version, then flags, then the actual condition.